### PR TITLE
Cooking with clovers

### PR DESCRIPTION
**Pull request by Hinaichigo**
This add three cooking recipes that use clovers:

**clover cream pie:** Like pumpkin pie but with clovers.
**clover con carne:** Clovers over steak with butter.
**tontes de pelouse:** A gourmet dish.

And makes it so when you cook something with an ingredient that has luckiness, the luckiness transfers to the resulting dish. If you slice something with luckiness, the luckiness will be divided up into the slices.
I got every case that I can think of; microwave, food press, candy machine, spaghetti, sandwich, etc. but please let me know if there's something I missed.

Also makes capitalization a little more consistent across some foods.

:cl:
 * rscadd: Added three clover-related recipes: clover cream pie, clover con carne, and tontes de pelouse.
 * rscadd: Lucky ingredients make for lucky dishes.

### CODE REVIEW

**Comment by Hinaichigo** [comment]
Note that luck currently already transfers when using the deep fryer and grill, as the original item remains in those cases.

**Comment by kacurasuo** [comment]
Does this mean that you can cook a clover cream pie with a 0 leaf clover and gib someone?

**Comment by Eneocho** [comment]
Add a small chance for the dish to be luckier than the ingredients, maybe 1.2x luck.

**Comment by Hinaichigo** [comment]
> Does this mean that you can cook a clover cream pie with a 0 leaf clover and gib someone?

Yep you can make unlucky clover cream pies that will make people unlucky. As to harming them... you could have them slip on a puddle and stab themselves, or maybe have them lose at gambling or russian roulette. 

> Add a small chance for the dish to be luckier than the ingredients, maybe 1.2x luck.

One thing is that that would allow for infinite luck when recursive food is enabled, especially if luck played a role.

**Comment by Eneocho** [comment]
I mean when the thing gets made out of the microwave so the luck stays non-recursive. You make your 5 leaf clover pie and got lucky, now it has 120 luck instead of 100 or make your 0 leaf clover pie and got lucky, now it has -12000 luck instead of -10000.

**Comment by Hinaichigo** [comment]
The idea of that would be to allow for multiplicative luck farming via:
...clover harvesting -> cook clover recipes -> eat them - > clover harvesting...
?

**Comment by Eneocho** [comment]
I mean, you can already just eat the regular clovers to stack your luck, making it so cooked clover food is luckier still would give an incentive to cook it.

**Comment by kacurasuo** [comment]
not everything in the game needs rng eneocho

**Comment by Eneocho** [comment]
> not everything in the game needs rng eneocho

To be honest I hate RNG, I'm just trying to argue in favour of having the cooked clover food be luckier than the ingredients you used. Having it on a chance would encourage cooking the stuff, since you wouldn't lose the luck when doing so, even if you didn't get lucky, you'd get food that's just as lucky as the clovers, nothing more.

**Comment by kacurasuo** [comment]
> I'm just trying to argue in favour of having the cooked clover food be luckier than the ingredients you

I don't mind this at all, so why not just propose it ALWAYS having more luck?

**Comment by Eneocho** [comment]
>I don't mind this at all, so why not just propose it ALWAYS having more luck?

Hinaichigo has tweaked the luck a few times and making them increase on cooking felt a bit much if you considered that, suggesting it to be under a chance seemed like a good middle ground. I'm all for cooked foods being luckier than the ingredients by some multiplier, since lesser clover foods could still feed the crew (and boost their luck instead of hoarding it) while the botanist farms for the big boy numbers

**Comment by Hinaichigo** [comment]
We'd have to assume that recursive food won't be turned on or else only have it happen with microwave food, but the latter seems a little inconsistent; for example a four-leaf clover-filled donut would be at the base luckiness but a clover cream pie would have extra luck. I'm not totally sure I like that parity-break.

**Comment by Eneocho** [comment]
Well, there's already a break between cooked food and processed food. For example, candy machine choco-coins do NOT work as coins, while properly cooked microwave choco coins do.
And balancing the idea with recursive food in mind is kinda weird, it's a feature that's only turned on maybe once every 1000 rounds? If that? I've only seen recursive foods enabled 2 or 3 times, the deepfry anything button is pressed way more. You don't really gotta worry about it. But if it worries you so much you could just add a flag that disables the extra luck if already processed or if recursive food is enabled.

**Comment by Hinaichigo** [comment]
> For example, candy machine choco-coins do NOT work as coins, while properly cooked microwave choco coins do.

In which situations do they not function properly as coins?
Mainly vending machines right? I wonder if letting them be used in a vending machine would be too powerful.

**Comment by Eneocho** [comment]
> > For example, candy machine choco-coins do NOT work as coins, while properly cooked microwave choco coins do.
> 
> In which situations do they not function properly as coins? Mainly vending machines right? I wonder if letting them be used in a vending machine would be too powerful.

Ye vending machines, and I think the HURT intent coin screwdriving too.

**Comment by Hinaichigo** [comment]
Also as railgun ammo apparently.

**Comment by Eneocho** [comment]
> Also as railgun ammo apparently.

Coins are the preferred railgun ammo actually. Diamond coins will FUCK YOU UP while a metal rod will maybe hurt.

File: code/datums/recipe.dm [hunk 1]
```diff
@@ -133,6 +133,11 @@
 				O.reagents.del_reagent(r_r)
 			//Transfer any reagents found in the object, to the resulting object
 			O.reagents.trans_to(result_obj, O.reagents.total_volume)
+		//Transfer any luckiness from the ingredients, to the resulting object
+		var/obj/item/I = O
+		var/obj/item/result_item = result_obj
+		if(I.luckiness)
+			result_item.luckiness += I.luckiness
```
Comment (Kurfursten) [inline code/datums/recipe.dm]: this will runtime when you cook a non-item food like the pop out cake

Comment (Hinaichigo) [reply code/datums/recipe.dm]: Added a check for both the ingredient and the dish.

**Comment by Eneocho** [comment]
Should've probably changed the name of the clover cream pie to clover pie, since the "cream" part implies it is throwable, like the banana cream pie. Could be implemented as a throwable pie though.

**Pull request merged**